src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -1446,12 +1446,24 @@ because xbuild doesn't support framework reference assemblies.
     _PrepareAssemblies;
     _PrepareNativeAssemblySources;
     $(_AfterPrepareAssemblies);
+    _GetGenerateJavaStubsInputs;
   </_GenerateJavaStubsDependsOnTargets>
 </PropertyGroup>
 
+<Target Name="_GetGenerateJavaStubsInputs">
+  <ItemGroup>
+    <_GenerateJavaStubsInputs Include="@(_AndroidMSBuildAllProjects)" />
+    <_GenerateJavaStubsInputs Include="$(_ResolvedUserAssembliesHashFile)" />
+    <_GenerateJavaStubsInputs Include="@(_ResolvedUserMonoAndroidAssemblies)" />
+    <_GenerateJavaStubsInputs Include="$(_AndroidManifestAbs)" />
+    <_GenerateJavaStubsInputs Include="$(_AndroidBuildPropertiesCache)" />
+    <_GenerateJavaStubsInputs Include="@(AndroidEnvironment);@(LibraryEnvironments)" Condition=" '$(EmbedAssembliesIntoApk)' == 'true' " />
```

Comment:
If we introduce a private flag `$(_AndroidFastDeployEnvironmentFiles)` for https://github.com/xamarin/monodroid/pull/1547, could we also check it here?

`@(_GenerateJavaStubsInputs)` should contain env var files if `$(_AndroidFastDeployEnvironmentFiles)` is `false`.

Reply:
How is this different for `EmbedAssembliesIntoApk` unless the idea is to allow us to override the behaviour?

Reply:
`$(_AndroidFastDeployEnvironmentFiles)` would let you keep using Fast Deployment, and just not Fast Deploy the environment files. It's just an extra a safety flag if something goes wrong.